objects/src/accounts/code/mod.rs

```diff
@@ -92,12 +100,16 @@ impl AccountCode {
     /// - Compilation of the provided source code fails.
     /// - The number of procedures exported from the provided library is smaller than 1 or greater
     ///   than 256.
-    pub fn compile(source_code: impl Compile, assembler: Assembler) -> Result<Self, AccountError> {
+    pub fn compile(
+        source_code: impl Compile,
+        assembler: Assembler,
+        is_faucet: bool,
+    ) -> Result<Self, AccountError> {
```

Comment:
I would probably keep this as it was, but would add `compile_faucet()` constructor.

Reply:
I would personally keep `compile` with the faucet check as is for the following reasons:
- `Compile` is supposed to be agnostic to the code that you are trying to compile be it a wallet, a faucet or any other interface hence it doesn't seem to make sense for me to have a `compile_x` and `compile_y` function
- The `is_faucet` flag is only temporary, until the assembler support storage offsets, hence a potential `compile_faucet` procedure would achieve the same as this flag with a lot more code and would also need to be removed.
- In the future I believe that we will need to remove this concept of faucet, will try to do some research on that

Reply:
@bobbinth let me know what you think.

Reply:
In terms of breaking changes, both approaches result in breaking changes in the future. My thinking was that introducing `compile_faucet()` would result in fewer code changes as we primarily deal with accounts, not faucets - but I'm fine keeping the flag as well.